File: packages/node/iii-example/src/iii.ts
import { init } from '@iii-dev/sdk'
import { version } from '../package.json'

// Engine WebSocket URL - used for both III and telemetry
const engineWsUrl = process.env.III_BRIDGE_URL ?? 'ws://localhost:49134'

export const iii = init(engineWsUrl, {
  otel: {
    enabled: true,
    serviceName: 'iii-example',
    metricsEnabled: true,
    serviceVersion: version,
    reconnectionConfig: {
      maxRetries: 10,
    },
    metricsExportIntervalMs: 10000,
  },
})

File: packages/python/iii-example/src/iii.py
import os

from iii import III, InitOptions, init

engine_ws_url = os.environ.get("III_BRIDGE_URL", "ws://localhost:49134")

_iii: III | None = None


def init_iii() -> III:
    global _iii
    if _iii is None:
        _iii = init(
            address=engine_ws_url,
            options=InitOptions(
                worker_name="iii-example",
                otel={"enabled": True, "service_name": "iii-example"},
            ),
        )
    return _iii


def get_iii() -> III:
    if _iii is None:
        raise RuntimeError("III client is not initialized. Call init_iii() from an active async context first.")
    return _iii
